crates/zed/src/languages/scala/outline.scm

```diff
@@ -0,0 +1,27 @@
+(class_definition
```

Comment:
The original PR by @hmemcpy (#6982) contained `overrides.scm` as well and the WIP branch from @bishabosha (https://github.com/bishabosha/zed/tree/feature-add-scala) had `locals.scm`. Would it make sense to integrate these too (sorry, I'm not really familiar with how Zed's treesitter configuration works)? :slightly_smiling_face:

Reply:
From https://github.com/mprihoda/zed/blob/aeb9e97138d332643d1d5b74584a222a709212a0/crates/language/src/language.rs#L761 it seems that locals.scm would not be used. Overrides should, but I'm not sure what to use it for, I have to figure it out yet.

The queries in current state seem to be working quite fine, though I'm sure there is a lot to be improved upon.

Reply:
I initially followed the comment in the code about the steps regarding adding a new language support, it mentioned copying the SC files from the language's tree-sitter repo. The Scala one had those, so I copied them as-is, but haven't checked if they were actually used by Zed.